src/ripple/app/tx/impl/SetSignerList.cpp

```diff
@@ -281,23 +283,23 @@ SetSignerList::removeSignersFromLedger (Keylet const& accountKeylet,
     if (!signers)
         return tesSUCCESS;
 
-    STArray const& actualList = signers->getFieldArray (sfSignerEntries);
+    STArray const& actualList = signers-> getFieldArray (sfSignerEntries);
     int const removeFromOwnerCount = ownerCountDelta (actualList.size()) * -1;
 
     // Remove the node from the account directory.
     auto const hint = (*signers)[sfOwnerNode];
 
-    auto viewJ = ctx_.app.journal ("View");
-    TER const result  = dirDelete(ctx_.view(), false, hint,
-        ownerDirKeylet, signerListKeylet.key, false, (hint == 0), viewJ);
-
-    if (result == tesSUCCESS)
-        adjustOwnerCount(view(),
-            view().peek(accountKeylet), removeFromOwnerCount, viewJ);
+    if (ctx_.view().dirRemove(
+            ownerDirKeylet, hint, signerListKeylet.key, false))
```

Comment:
I think this should also return `tefBAD_LEDGER` if `dirRemove` fails.

Reply:
Agreed. Setting consistency with this behavior of always returning `tefBAD_LEDGER` if any of the removals fail will make it easier to reason about.